build/tests/04-bins.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+if [ -n "${VERBOSE:-}" ]; then
+    set -x
+fi
+
+source .requirements
+source build/tests/util.sh
+
+msg_test "Check if Kong provided cURL is executable"
+assert_exec 0 'root' "test -x /usr/local/kong/bin/curl"
+
+msg_test "Check if Kong provided cURL runs correctly"
+assert_exec 0 'root' "/usr/local/kong/bin/curl --version"
+
+msg_test "Check if Kong provided cURL can do HTTP2"
+assert_exec 0 'root' "/usr/local/kong/bin/curl -sI https://curl.se -o/dev/null -w '%{http_version}\n' | grep 2"
```

Comment:
:thinking: This is bound to be flaky at some point, but I'm struggling to think of A Better Way :tm:...

There _is_ an `--http2` cli flag that (presumably) only shows up if curl was compiled with http2 support:

```
$ curl --help all | grep http2
     --http2              Use HTTP 2
     --http2-prior-knowledge Use HTTP 2 without HTTP/1.1 Upgrade
```

We could check for that? Not sure if it's sufficient or not.

At very least I think adding `--http2` to this check would be wise since it's a bit more explicit (what if for some reason curl.se's load balancer doesn't auto-upgrade us to http2 for some reason?)

```suggestion
assert_exec 0 'root' "/usr/local/kong/bin/curl --http2 -sI https://curl.se -o/dev/null -w '%{http_version}\n' | grep 2"
```

None of this is a big deal/a blocker, just pointing out things :shrug:

Reply:
Built a curl locally without http2 to validate that it wouldn't have that flag available. It ended up having it, but using it produced:
```
curl: option --http2: the installed libcurl version doesn't support this
curl: try 'curl --help' for more information
<exit 2>
```

Which is slightly un helpful.

I think the best way to validate that our supplied curl has http2 support (without actually using it) is to parse the "Features:..." field of `--version`. So I've gone with that.